File: task_equivalents/llm_translations/gemini/base/r/data_visualization/01_scatter_trend_grouped.R
Auto <- read.csv('data/csv/Auto.csv', na.strings = '?')
Auto <- na.omit(Auto)

X <- Auto$horsepower
Y <- Auto$mpg
model1_fit <- lm(Y ~ X)

a <- coef(model1_fit)[1]
b <- coef(model1_fit)[2]
yfit <- a + b * X

plot(X, Y, xlab = 'horsepower', ylab = 'mpg')
lines(X, yfit, col = 'red')

File: task_equivalents/llm_translations/gemini/base/python/data_visualization/01_scatter_trend_grouped.py
import pandas as pd
import numpy as np
import statsmodels.api as sm
import matplotlib.pyplot as plt

auto = pd.read_csv('data/csv/Auto.csv', na_values='?')
auto = auto.dropna()

n = len(auto)
X_col = auto['horsepower'].to_numpy().reshape(-1, 1)
Y = auto['mpg'].to_numpy().reshape(-1, 1)
X = np.column_stack((np.ones(n), X_col))

model1 = sm.OLS(Y, X).fit()

a = model1.params[0]
b = model1.params[1]
yfit = a + b * X[:, 1]

plt.scatter(X[:, 1], Y, facecolors='none', edgecolors='k')
plt.xlabel('horsepower')
plt.ylabel('mpg')
plt.plot(X[:, 1], yfit, color='red')
plt.show()